File: py/day01.py
import os

with open(os.path.dirname(os.path.realpath(__file__)) + "/../input/day01-input", "r") as f:
    input = [int(line) for line in f.readlines()]

part1 = 0
part2 = 0

for i in range(1, len(input)):
    if input[i] > input[i-1]:
        part1 += 1

wa = input[0] + input[1] + input[2]
wb = wa
for i in range(2, len(input) - 1):
    wb = wb - input[i-2] + input[i+1]
    if wb > wa:
        part2 += 1
    wa = wb

print("Day 1")
print("  Part 1:", part1)
print("  Part 2:", part2)

File: cpp/src/day01.cpp
#include <vector>
#include "aocutil.h"

using namespace std;

void day01()
{
	auto input = get_input_ints(1, false);
	int ilen = (int)input.size();
	int part1 = 0;
	for (int i = 1; i < ilen; i++) {
		if (input[i] > input[i-1]) {
			part1++;
		}
	}

	int part2 = 0;
	int last = input[0] + input[1] + input[2];
	for (int i = 2; i < ilen - 1; i++) {
		int curr = input[i - 1] + input[i] + input[i + 1];
		if (curr > last) {
			part2++;
		}
		last = curr;
	}

	//printDayResult(1, 1, part1);
	//printDayResult(1, 2, part2);
	printDayResults(1, part1, part2);
}
